guide/message_transformation/message_transformation.bal

```diff
@@ -0,0 +1,370 @@
+                log:printError("Error while content reading", err = e) };
+            }
+        }
+
+    }
+}
+
+//The content enricher service
+service<http:Service> enricher bind contentenricherEP {
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/"
+    }
+    enricher(endpoint caller, http:Request validatedreq) {
+        http:Request enrichedreq = validatedreq;
+        var jsonMsg = validatedreq.getJsonPayload();
+        match jsonMsg {
+            json msg => {
+                //get the student's ID value
+                int Idvalue = check <int>msg["id"];
+                //select details from the data table according to the student's ID
+                var selectRet = testDB->select("SELECT * FROM student", ());
+                table dt;
+                match selectRet {
+                    table tableReturned => dt = tableReturned;
+                    error e => io:println("Select data from student table failed: "
+                            + e.message);
+                }
+                //convert the details to a jason file
+                io:println("\nConvert the table into json");
+                var jsonConversionRet = <json>dt;
+                match jsonConversionRet {
+                    json jsonRes => {
+                        //set student's details to the global variable
+                        payload1= untaint jsonRes;
+                        io:println(payload1);
+                    }
+                    error e => io:println("Error in student table to json conversion");
+                }
+                //drop the student details table
+                var ret = testDB->update("DROP TABLE student");
+                handleUpdate(ret, "Drop table student");
+                //select student's results from the student results data table, according to the student's ID
+                var selectRet1 = testDB1->select("select Com_Maths,Physics,Chemistry from StudentDetails where ID = ?", (), Idvalue);
+                table dt1;
+                match selectRet1 {
+                    table tableReturned => dt1 = tableReturned;
+                    error e => io:println("Select data from StudentDetails table failed: "
+                            + e.message);
+                }
+                //convert the details to a jason file
+                io:println("\nConvert the table into json");
+                var jsonConversionRet1 = <json>dt1;
+                match jsonConversionRet1 {
+                    json jsonRes1 => {
+                        //set student's result details to the global variable
+                        payload2= untaint jsonRes1;
+                    }
+                    error e => io:println("Error in StudentDetails table to json conversion");
+                }
+                testDB.stop();
+
+                //JSON to JSON conversion to the selected details
+                //define new jason variable
+                json pay = payload1[0];
```

Comment:
shall we consider renaming `pay` to something more meaningful? 

also, typo in the comment `jason`.

Reply:
Done